<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="utf-8" />
    <title></title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" href="gem_navbar.css">
</head>



<body>

    <!-- Navigation bar -->
    <nav class="navbar navbar-default">
        <div class="container-fluid">
            <div class="navbar-header">
                <a class="navbar-brand" href="index.html">GEM</a>
            </div>

            <ul class="nav navbar-nav">
                <li><a href="index.html">Home</a></li>
                <li><a href="installation.html">Installation</a></li>
                <li><a href="file_formats.html">File Formats</a></li>
                <li><a href="parameters.html">Parameters</a></li>
                <li><a href="examples_and_debugging.html">Examples and Debugging</a></li>
                <li><a href="resource_calculator.html">Resource Calculator</a></li>
            </ul>
        </div>
    </nav>



    <!-- Section of options -->
    <br />
    <h2 align="center">Input/Output Files</h2>

    <p align="center">
        <a href="#Genotype_Files">Genotype Files</a> |
        <a href="#Phenotype_File">Phenotype File</a> |
        <a href="#Output_File">Output File</a> |
        <a href="#Sample_File">Sample File</a>
    </p>

    <br />
    <br />
    <br />




    <!-- Option Table -->
    <table id="option_table" style="margin-left:200px;table-layout:fixed;width:650px">

        <!-- Genotype Files -->
        <tr class="thead">
            <td colspan="1" , style="width:200px"><h3 , id="Genotype_Files"><u>Genotype Files</u></h3></td>
        </tr>
        <tr class="thead2">
            <td colspan="1"><h4>.bgen</h4></td>
        </tr>
        <tr>
            <td width="100">
                <div style="line-height: 1.5em">
                    GEM supports version 1.1, 1.2 and 1.3 binary GEN (BGEN) genotype files described <a href="https://www.well.ox.ac.uk/~gav/bgen_format/spec/latest.html"><u>here</u></a>.
                    <br /><br />A .sample file should be included with the .bgen file if there is not a sample identifier block within the .bgen file. This can be done by including the <code>--sample</code> flag.
                    <br /><br />In GEM versions &ge; 1.1, if the .sample file is included and the .bgen file does contain a sample identifier block, GEM will prioritize the .sample file over the sample identifier block for sample identifier matching.
                </div>
            </td>

        </tr>

        <tr class="spacer"><td colspan="1">&nbsp;</td></tr>
        <tr class="spacer"><td colspan="1">&nbsp;</td></tr>
        <tr class="spacer"><td colspan="1">&nbsp;</td></tr>
        <tr class="spacer"><td colspan="1">&nbsp;</td></tr>
        <tr class="spacer"><td colspan="1">&nbsp;</td></tr>


        <!-- Phenotype File -->
        <tr class="thead">
            <td colspan="1"><h3 , id="Phenotype_File"><u>Phenotype File</u></h3></td>
        </tr>
        <tr class="thead2">
            <td colspan="1"><h4>.pheno</h4></td>
        </tr>
        <tr>
            <td>
            <div style="line-height: 1.5em">
                 The phenotype file does not need to have the .pheno extension and can be a different type of extension (.csv, .txt, etc...). However, the appropriate delimiter of the phenotype file should be specified using the <code>--delim</code> parameter. The default delimiter for phenotype files in GEM is , (comma-seperated). <br />
                 <br />A phenotype file should contain a sample identifer column, along with the phenotype, covariate, interactive covariate and exposure columns. The ordering of the columns does not matter. <br />The first row of the phenotype file should contain the header name for each column.
            </div>
            </td>
        </tr>

        <tr class="spacer"><td colspan="1">&nbsp;</td></tr>
        <tr class="spacer"><td colspan="1">&nbsp;</td></tr>
        <tr class="spacer"><td colspan="1">&nbsp;</td></tr>
        <tr class="spacer"><td colspan="1">&nbsp;</td></tr>
        <tr class="spacer"><td colspan="1">&nbsp;</td></tr>


        <!--Output File-->
        <tr class="thead">
            <td colspan="1"><h3 , id="Output_File"><u>Output File</u></h3></td>
        </tr>
        <tr class="thead2">
            <td colspan="1"><h4>.out</h4></td>
        </tr>
        <tr>
            <td>
                <div style="line-height: 1.5em">
                    The <code>--out</code> flag should contain the full path and extension to where GEM output results. Otherwise, GEM will use <code>gem.out</code> as the default output file name and results will most likely be located in the directory where GEM was executed.
                 </div>
                    
                <br /><br /><br />Below is a table describing the column names in the output file. <br /><br />
                <table id="option_table" style="table-layout:fixed;">

                        <!-- General Options -->
                        <tr class="thead">
                            <td colspan="1" , style="margin-left=20px;width:225px"></td>
                        </tr>
                        <tr class="thead2" , style="background-color: #f2f2f2">
                            <td colspan="1" , align="center"><h4>Header</h4></td>
                            <td colspan="1" , align="center"><h4>Description</h4></td>
                        </tr>
                        <tr><td height="35" , align="center">SNPID</td><td height="35" , align="center">The SNP identifier as retrieved from the BGEN file.</td></tr>
                        <tr style="background-color: #f2f2f2"><td height="35" , align="center">rsID</td><td height="35" , align="center">The reference SNP ID number.</td></tr>
                        <tr><td height="35" , align="center">CHR</td><td height="35" , align="center">The chromosome of the SNP.</td></tr>
                        <tr style="background-color: #f2f2f2"><td height="35" , align="center">POS</td><td height="35" , align="center">The physical position of the SNP.</td></tr>
                        <tr><td height="35" , align="center">Allele1</td><td height="35" , align="center">The first allele in the BGEN file.</td></tr>
                        <tr style="background-color: #f2f2f2"><td height="55" , align="center">Allele2</td><td height="55" , align="center">The second allele in the BGEN file <br />that is counted in association testing.</td></tr>
                        <tr><td height="35" , align="center">AF</td><td height="35" , align="center">The allele frequency of the second allele in the BGEN file.</td></tr>
                        <tr style="background-color: #f2f2f2"><td height="35" , align="center">Beta_Marginal</td><td height="35" , align="center">The coefficient estimate for the marginal genetic effect.</td></tr>
                        <tr><td height="55" , align="center">Var_Beta_Marginal</td><td height="35" , align="center">The variance associated with the marginal genetic effect estimate.</td></tr>
                        <tr style="background-color: #f2f2f2"><td height="55" , align="center">Beta_Interaction_k</td><td height="35" , align="center">The coefficient estimate for the kth exposure term, <br /> where <code>k = {1..length(--exposure-names)}</code>.</td></tr>
                        <tr><td height="55" , align="center">Var_Beta_Interaction_k_ j</td><td height="35" , align="center">The variance associated with the kth and jth exposure term, <br /> where <code> j = {1..length(--exposure-names)}</code>.</td></tr>
                        <tr style="background-color: #f2f2f2"><td height="35" , align="center">P_Value_Marginal</td><td height="35" , align="center">Marginal genetic effect p-value.</td></tr>
                        <tr><td height="35" , align="center">P_Value_Interaction</td><td height="35" , align="center">Interaction effect p-value.</td></tr>
                        <tr style="background-color: #f2f2f2"><td height="35" , align="center">P_Value_Joint</td><td height="35" , align="center">Joint test p-value (K+1 degrees of freedom test of genetic effect).</td></tr>
                </table>
            </td>
        </tr>


        <tr class="spacer"><td colspan="1">&nbsp;</td></tr>
        <tr class="spacer"><td colspan="1">&nbsp;</td></tr>
        <tr class="spacer"><td colspan="1">&nbsp;</td></tr>
        <tr class="spacer"><td colspan="1">&nbsp;</td></tr>
        <tr class="spacer"><td colspan="1">&nbsp;</td></tr>


        <!--Sample File-->
        <tr class="thead">
            <td colspan="1"><h3 , id="Sample_File"><u>Sample File</u></h3></td>
        </tr>
        <tr class="thead2">
            <td colspan="1"><h4>.sample</h4></td>
        </tr>
        <tr>
            <td>
                <div style="line-height: 1.5em">
                    A .sample file is required when the BGEN file does not include a sample identifier block. <br />Include a .sample file with the <code>--sample</code> option. <br />
                    <br />Formats for the .sample file should follow QCTOOLv2 format described <a href="https://www.well.ox.ac.uk/~gav/qctool_v2/documentation/sample_file_formats.html"><u>here</u></a>.
                    <br />The first column in the .sample file should always be the ID column used for ID matching with the phenotype file.
                    <br /><br /> In GEM versions &ge; 1.1, the .sample file will be prioritized over the sample identifier block in the BGEN file for sample identifier matching.
                </div>
            </td>
        </tr>




        <tr class="spacer"><td colspan="1">&nbsp;</td></tr>
        <tr class="spacer"><td colspan="1">&nbsp;</td></tr>
        <tr class="spacer"><td colspan="1">&nbsp;</td></tr>
        <tr class="spacer"><td colspan="1">&nbsp;</td></tr>
        <tr class="spacer"><td colspan="1">&nbsp;</td></tr>

    </table>

    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />
    <br />

    <script>
            function myFunction() {
                // Declare variables
                var input, filter, table, tr, td, i, txtValue;
                input  = document.getElementById("myInput");
                filter = input.value.toUpperCase();
                table  = document.getElementById("option_table");
                tr = table.getElementsByTagName("tr");

                // Loop through all table rows, and hide those who don't match the search query
                for (i = 0; i < tr.length; i++) {
                    td = tr[i].getElementsByTagName("td")[0];
                    if (td) {
                        txtValue = td.textContent || td.innerText;
                        if (txtValue.toUpperCase().indexOf(filter) > -1) {
                            tr[i].style.display = "";
                        } else {
                            tr[i].style.display = "none";
                        }
                    }
                }
            }
    </script>
</body>


</html>


