<!DOCTYPE html>
<html lang = "en">

<head>
    <meta charset="utf-8" />
    <title>GEM Website</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <link rel="stylesheet" href="gem_navbar.css">
</head>
<body>



    <!-- Navigation bar -->
    <nav class="navbar navbar-default">
        <div class="container-fluid">
            <div class="navbar-header">
                <a class="navbar-brand" href="index.html">GEM</a>
            </div>

            <ul class="nav navbar-nav">
                <li><a href="index.html">Home</a></li>
                <li><a href="installation.html">Installation</a></li>
                <li><a href="file_formats.html">File Formats</a></li>
                <li><a href="parameters.html">Parameters</a></li>
                <li><a href="examples_and_debugging.html">Examples and Debugging</a></li>
                <!--<li><a href="resource_calculator.html">Resource Calculator</a></li>-->
            </ul>
        </div>
    </nav>

    <br />



    <h4 align="center"></h4>
    <div id="tester" style="width:800px;height:500px;margin-left:125px;">
        <div style="font-size:30px;margin-left:340px">
            <b>
                GEM
            </b>
        </div>

        <p align="center" style="font-size:16px;">
            <u>G</u>ene-<u>E</u>nvironment interaction for <u>M</u>illions of samples
        </p>

        <br />

        <p style="font-size:15px;line-height:1.5;" align="left">
            GEM (Gene-Environment interaction analysis for Millions of samples) is a software program for large-scale gene-environment interaction testing in samples from unrelated individuals. It enables genome-wide association studies in up to millions of samples while allowing for multiple exposures, control for genotype-covariate interactions, and robust inference.
        </p>
        <br />

        <br />
        <br />
        <br />
        <br />

        <style>
            #update_block {
                padding: 0 .2em;
                background: white;
            }
        </style>


        <h2><u>Recent Updates</u>: </h2>
        <h3><a href="https://github.com/large-scale-gxe-methods/GEM/releases/tag/v1.3"><u>Version 1.3</u></a> - April 7, 2021: </h3>
        <div id="update_block" style="line-height: 1.5em">
            <table>
                <tr>
                    <td height="25">
                        03/18/2021 - Add a new flag (--output-style) to modify which summary statistics should be included in the the output file. <br />
                        &emsp;&emsp;&emsp;&emsp;&emsp;&nbsp;&nbsp;&nbsp; Column names now include the exposure and interaction covariate names instead of numbers. <br />
                        02/02/2021 - The --exposure-names flag is now optional. If no exposures are specified, GEM will run a G-only model. <br />
                        &emsp;&emsp;&emsp;&emsp;&emsp;&nbsp;&nbsp;&nbsp; Covariates (not of interest) can still be adjusted for using --covar-names flag. <br />
                    </td>
                </tr>
            </table>
        </div>

        <br />
        <br />
        <br />
        <h3><a href="https://github.com/large-scale-gxe-methods/GEM/releases/tag/v1.2"><u>Version 1.2</u></a> - January 22, 2021: </h3>
        <div id="update_block" style="line-height: 1.5em">
            <table>
                <tr>
                    <td height="25">
                        12/15/2020 - Fix issue to allow for space and tab delimited phenotype files. <br />
                        12/09/2020 - Allow for centering and scaling of exposures and covariates. <br />
                        12/02/2020 - Update calculations for model-based joint test <br />
                        11/23/2020 - Update calculations for robust joint test <br />
                        11/17/2020 - Output covariance, coefficients and standard errors to the log.  <br />
                        10/13/2020 - Change Allele1 and Allele2 in outfile file to Non_Effect_Allele and Effect_Allele.  <br />
                        10/13/2020 - Fix bug when phenotype is binary and there are missing genotypes.  <br />
                        09/22/2020 - Support PGEN/BED files.  <br />
                    </td>
                </tr>
            </table>
        </div>

        <br />
        <br />
        <br />

        <h3><a href="https://github.com/large-scale-gxe-methods/GEM/releases/tag/v1.1"><u>Version 1.1</u></a> - July 21, 2020: </h3>
        <div id="update_block" style="line-height: 1.5em">
            <table>
                <tr><td height="25">05/18/2020 - Allow GEM to subset the BGEN file based on a list of variants to include for analysis. <code>--include-snp-file</code> <br /></td></tr>
                <tr>
                    <td height="25">
                        06/26/2020 - Use matrix operation to adjust for covariates instead of for-loop. <br />
                        &emsp;&emsp;&emsp;&emsp;&emsp;&nbsp;&nbsp;&nbsp; Use the <a href="https://github.com/ebiggers/libdeflate"><u>libdeflate</u></a> package for faster zlib decompression of BGEN genotype blocks <br />
                        &emsp;&emsp;&emsp;&emsp;&emsp;&nbsp;&nbsp;&nbsp; Compile GEM with -O2 (optimizer flag) <br />
                        &emsp;&emsp;&emsp;&emsp;&emsp;&nbsp;&nbsp;&nbsp; Prioritize BGEN sample file over the BGEN sample identifier block <br />
                        &emsp;&emsp;&emsp;&emsp;&emsp;&nbsp;&nbsp;&nbsp; Error if phenotype (--pheno-name) is also included as an exposoure or covariate <br />
                        06/30/2020 - Support BGEN v1.1, v1.2 and v1.3 uncompressed genotype blocks.  <br />
                        07/15/2020 - Fix major printing bug.  <br />
                        07/21/2020 - Handle missing genotypes in BGEN files.  <br />
                    </td>
                </tr>
            </table>
        </div>


        <br />
        <br />
        <br />
        <br />

        <h2><u>Citation</u>: </h2>
        <div>
            Westerman, K. E., Pham, D. T., Hong, L., Chen, Y., Sevilla-Gonz&aacute;lez, M., Sung, Y. J., Sun, Y. V., Morrison, A. C., Chen, H., Manning, A. K. Scalable and flexible gene-environment interaction analysis in millions of samples. (2020). bioRxiv. <a href="https://doi.org/10.1101/2020.05.13.090803">https://doi.org/10.1101/2020.05.13.090803 </a>.
        </div>

        <br />
        <br />
        <br />
        <br />

        <h2><u>License</u>: </h2>
        <div style="font-size:15px;">
            GEM : Gene-Environment interaction analysis for Millions of samples <br />
            Copyright (C) 2018-2021  Liang Hong, Han Chen, Duy Pham <br />
            <br />
            This program is free software: you can redistribute it and/or modify
            it under the terms of the GNU General Public License as published by
            the Free Software Foundation, either version 3 of the License, or
            (at your option) any later version.
        </div>
        <br />
        <img src="http://www.gnu.org/graphics/gplv3-88x31.png" alt="Trulli" width="100" height="25"> <a href="http://www.gnu.org/copyleft/gpl.html">GNU General Public License, GPLv3</a>
        <br />
        <br />
        <br />
        <br />
        <br />
        <br />
        <br />
        <br />
        <br />
        <br />
    </div>


</body>
</html>


